src/core/js/views/adaptView.js

```diff
@@ -30,185 +30,166 @@ define([
       this.preRender();
       this.render();
       this.setupOnScreenHandler();
-    },
+    }
 
-    preRender: function() {},
+    preRender() {}
 
-    postRender: function() {
+    postRender() {
       this.addChildren();
-    },
+    }
 
-    render: function() {
-      Adapt.trigger(this.constructor.type + 'View:preRender', this);
+    render() {
+      const type = this.constructor.type;
+      Adapt.trigger(`${type}View:preRender`, this);
 
-      var data = this.model.toJSON();
+      const data = this.model.toJSON();
       data.view = this;
-      var template = Handlebars.templates[this.constructor.template];
+      const template = Handlebars.templates[this.constructor.template];
       this.$el.html(template(data));
 
-      Adapt.trigger(this.constructor.type + 'View:render', this);
+      Adapt.trigger(`${type}View:render`, this);
 
-      _.defer(function() {
+      _.defer(() => {
         // don't call postRender after remove
         if (this._isRemoved) return;
 
         this.postRender();
-        Adapt.trigger(this.constructor.type + 'View:postRender', this);
-      }.bind(this));
+        Adapt.trigger(`${type}View:postRender`, this);
+      });
 
       return this;
-    },
+    }
 
-    setupOnScreenHandler: function() {
-      var onscreen = this.model.get('_onScreen');
+    setupOnScreenHandler() {
+      const onscreen = this.model.get('_onScreen');
 
       if (!onscreen || !onscreen._isEnabled) return;
 
       this.$el
         .addClass('has-animation')
-        .addClass(onscreen._classes + '-before');
+        .addClass(`${onscreen._classes}-before`);
 
-      this.$el.on('onscreen.adaptView', function (e, m) {
+      this.$el.on('onscreen.adaptView', (e, m) => {
 
         if (!m.onscreen) return;
 
-        var minVerticalInview = onscreen._percentInviewVertical || 33;
+        const minVerticalInview = onscreen._percentInviewVertical || 33;
 
         if (m.percentInviewVertical < minVerticalInview) return;
 
-        this.$el.addClass(onscreen._classes + '-after' || 'onscreen').off('onscreen.adaptView');
+        const classes = onscreen._classes ? `${onscreen._classes}-after` : 'onscreen';
```

Comment:
`onscreen._classes + '-after' || 'onscreen'` how does this work?

Reply:
onscreen is not used, leftover code from rewrite